opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml

```diff
@@ -338,6 +338,33 @@
       <downtime begin="0" end="432000000" interval="300000"/><!-- 5m, 0, 5d -->
       <downtime begin="432000000" interval="3600000"/><!-- 1h, 5d -->
    </package>
+
+   <package name="device-config">
+      <filter>IPADDR != '0.0.0.0'</filter>
+      <rrd step="300">
+         <rra>RRA:AVERAGE:0.5:1:2016</rra>
+         <rra>RRA:AVERAGE:0.5:12:1488</rra>
+         <rra>RRA:AVERAGE:0.5:288:366</rra>
+         <rra>RRA:MAX:0.5:288:366</rra>
+         <rra>RRA:MIN:0.5:288:366</rra>
+      </rrd>
+      <service name="DeviceConfig-Default" interval="30000" user-defined="false" status="on">
+         <parameter key="retry" value="3"/>
+         <parameter key="timeout" value="3000"/>
+         <parameter key="rrd-repository" value="${install.share.dir}/rrd/response"/>
+         <parameter key="rrd-base-name" value="device-config"/>
+         <parameter key="config-type" value="Default"/>
```

Comment:
Maybe we should add comments on the other necessary parameters here:
* script
* user
* password

BTW: How would we configure a multi-line script. The line breaks are important.

Having the script and credentials in the service configuration means that there may be more than one service definition for a certain config type. Is the mapping configType -> service unambiguous at least for a certain IP? Otherwise we have to rethink how the service is selected in case of a manual trigger.

Reply:
If we have multi-line script, we should provide file path for that script. 
Since other parameters will mostly come from metadata, I haven't added anything but we can definitely add comment here.

Reply:
Where would that file path point to an absolute path on the file system?

Reply:
For adding the script right into the service configuration would could also use indexed keys, e.g.:
```
         <parameter key="script.1" value="send: ..."/>
         <parameter key="script.2" value="send: atftp --put --local-file somefile.cfg --remote-file ${filename} ${tftpServerIp} ${tftpServerPort}"
         <parameter key="script.3" value="await: ..."/>
```

Reply:
> For adding the script right into the service configuration would could also use indexed keys, e.g.:
> 
> ```
>          <parameter key="script.1" value="send: ..."/>
>          <parameter key="script.2" value="send: atftp --put --local-file somefile.cfg --remote-file ${filename} ${tftpServerIp} ${tftpServerPort}"
>          <parameter key="script.3" value="await: ..."/>
> ```

It's easier to configure a filename in the UI. But we may have to parse that script and load it in an object before sending it to Minion since minion doesn't have access to files on OpenNMS.